pkg/secrets/aws_sm_client.go

```diff
@@ -0,0 +1,46 @@
+package secrets
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+)
+
+type AwsSecretsManagerClient interface {
+	FetchSecret(region string, secretName string) (*secretsmanager.GetSecretValueOutput, error)
+}
+
+type AwsSecretsManagerClientImpl struct{}
+
+func NewAwsSecretsManagerClient() AwsSecretsManagerClient {
+	return &AwsSecretsManagerClientImpl{}
+}
+
+func (a *AwsSecretsManagerClientImpl) FetchSecret(region string, secretName string) (*secretsmanager.GetSecretValueOutput, error) {
+	sess, err := session.NewSession(&aws.Config{
```

Comment:
Shouldn't all this setup jazz be members of the Impl or part of the constructor, rather than in FetchSecret, or do we not expect the interface to grow over time?

Reply:
I only made this so I could mock it in my unit test.

The prior art in this project just had this entire function in the Decrypter file: https://github.com/armory/go-yaml-tools/blob/master/pkg/secrets/s3.go#L75-L84

but I wanted to be able to mock this function so I created a interface / client for it that just had this 1 object.

I am not objected to moving that logic into the "constructor"

Reply:
@dogonthehorizon I agree with you on this one, I will refactor this.
- [ ]  Refactor the client creation to the constructor and save it in the struct.